mediapipe/calculators/ovms/openvinoinferencecalculator.cc

```diff
@@ -333,12 +358,166 @@ class OpenVINOInferenceCalculator : public CalculatorBase {
     std::unique_ptr<tflite::Interpreter> interpreter_ = absl::make_unique<tflite::Interpreter>();
     bool initialized = false;
 
+    static bool ValidateOrderLists(std::set<std::string> calculatorTags, const google::protobuf::RepeatedPtrField<std::string>& order_list) {
+        // Get output_stream types defined in the graph
+        std::vector<std::string> inputTypes;
+        for (const std::string& tag : calculatorTags) {
+            std::vector<std::string> tokens = tokenize(tag, ':');
+            if (tokens.size() > 0) {
+                std::string inputType = tokens[0];
+
+                // Check if supported vector tag was used
+                for (const auto& supportedVectorTag : supportedVectorTags) {
+                    if ( startsWith(inputType, supportedVectorTag)){
+                        if (order_list.size() < 1)
+                        {
+                            LOG(ERROR) << "OpenVINOInferenceCalculator GetContract error. Order list is requiered for vector types: " << inputType;
+                            return false;
+                        }
+                    }
+                }
+            }
+        }
+
+        return true;
+    }
+
+    static bool ValidateTagToNames(std::set<std::string> calculatorTags, const google::protobuf::Map<std::string, std::string>& tags_to_names) {
+        // Get output_stream types defined in the graph
+        std::vector<std::string> inputTypes;
+        for (const std::string& tag : calculatorTags) {
+            std::vector<std::string> tokens = tokenize(tag, ':');
+            if (tokens.size() > 0) {
+                inputTypes.push_back(tokens[0]);
+            } else {
+                inputTypes.push_back(tag);
+            }
+        }
+
+        for (const auto& [key, value] : tags_to_names) {
+            bool nameMatch = false;
+
+            // Check if supported tag was used
+            for (const auto& supportedTag : supportedTags) {
+                if ( startsWith(key, supportedTag)){
+                    if (endsWith(key, "S") && !endsWith(supportedTag, "S"))
```

Comment:
Do we really want to implicitly treat all types that end with "S" with vector types?

Reply:
Those are only types from our supported list. The check is so that if key is found in supported tags it might be TENSOR or RENSORS, thus we have to double check it is a match.